partiql-eval/src/test/kotlin/org/partiql/eval/internal/PartiQLEvaluatorTest.kt

```diff
@@ -1024,13 +1025,146 @@ class PartiQLEvaluatorTest {
                     int32Value(3),
                 )
             ),
+            // TODO port `IS <boolean value>` tests to conformance tests
+            // IS TRUE
+            SuccessTestCase(
+                input = "TRUE IS TRUE;",
+                expected = boolValue(true)
+            ),
+            SuccessTestCase(
+                input = "FALSE IS TRUE;",
+                expected = boolValue(false)
+            ),
+            SuccessTestCase(
+                input = "NULL IS TRUE;",
+                expected = boolValue(false)
+            ),
+            SuccessTestCase(
+                input = "MISSING IS TRUE;",
+                expected = boolValue(false)
+            ),
+            SuccessTestCase(
+                input = "'foo' IS TRUE",
+                expected = Datum.missing(),
+                mode = Mode.PERMISSIVE()
```

Comment:
Something interesting is Postgres does

```
'1' IS TRUE  -- true
'foo' IS TRUE -- err!
```

Reply:
So in strict mode, `'foo' IS TRUE` should error (assuming https://github.com/partiql/partiql-lang-kotlin/issues/1680 gets fixed) since the expression for `IS <truth value>` expects a boolean value.

This is also consistent w/ what we do for the other boolean constructs like `AND`, `OR`, `NOT` (i.e. data type mismatch error in strict, missing in permissive).

Postgres allows certain strings as input to functions that expect booleans. From their [docs](https://www.postgresql.org/docs/current/datatype-boolean.html):
> The datatype input function for type `boolean` accepts these string representations for the “true” state:
> - true 
> - yes 
> - on 
> - 1 
> 
> and these representations for the “false” state: 
> 
> - false 
> - no 
> - off 
> - 0 
> 
> Unique prefixes of these strings are also accepted, for example `t` or `n`. Leading or trailing whitespace is ignored, and case does not matter.

```SQL
-- in Postgres
-- IS TRUE
SELECT 'no' IS TRUE -- false
SELECT 'foo' IS TRUE -- Query Error: invalid input syntax for type boolean: "foo"
-- IS FALSE
SELECT 'no' IS FALSE -- true
SELECT 'foo' IS FALSE -- Query Error: invalid input syntax for type boolean: "foo"
-- IS UNKNOWN
SELECT 'no' IS UNKNOWN -- false
SELECT 'foo' IS UNKNOWN -- Query Error: invalid input syntax for type boolean: "foo"
-- other boolean exprs
SELECT NOT 'no' -- true
SELECT NOT 'foo' -- Query Error: invalid input syntax for type boolean: "foo"
SELECT true AND 'no' -- false
SELECT true AND 'foo' -- Query Error: invalid input syntax for type boolean: "foo"
SELECT false OR 'no' -- false
SELECT false OR 'foo' -- Query Error: invalid input syntax for type boolean: "foo"
```